lesson17/script.js

```diff
@@ -0,0 +1,127 @@
+const slider = document.getElementById("js-slider");
+
+const createElement = (obj) => {
+    const element = document.createElement(obj.tagName);
+    if(obj.id){
+        element.id = obj.id;
+    }
+    element.className = obj.className;
+    element.textContent = obj.text;
+    element.src = obj.src;
+    element.alt = obj.alt;
+    return element;
+}
```

Comment:
IMO
That looks really nice!
This is the kind of image I was thinking of. What do you think?
In this way, I think we can handle whatever attributes are passed to us.

```suggestion
const createAttributedElements = (tag, attributes) => {
  const element = document.createElement(tag);
  Object.keys(attributes).forEach((attribute) =>
    element.setAttribute(attribute, attributes[attribute])
  );
  return element;
};
```

```js
const hoge = createAttributedElements("img", {
  id: "js-loading",
  className: "loading",
  src: "img/loading-circle.gif",
  alt: "ローディング"
});

```
result
```html
<img id="js-loading" classname="loading" src="img/loading-circle.gif" alt="ローディング"></img>
```

Reply:
I wanted to separate tags and attributes, so I think it's very good!
If I want text to have a proper name as below, how can you do it?
Since text is often not a simple string, I wanted to give the caller a proper name so that it could be identified as text.
```
const pageNumber = createAttributesElement("p",{
    class:"page-number"
    },
    text:"/${sliderData.length"
);
```

Reply:
What about code like this?

```
const createAttributedElements = ({tag,attrObj,text}) =>{
    const element = document.createElement(tag);
    Object.keys(attrObj).forEach((attribute) => {
        element.setAttribute(attribute, attrObj[attribute]);
    });
    if(text !== undefined) element.textContent = text;
    return element;
}
```
result
```
const hoge = createAttributedElements({
  tag:"img", 
  attrObj:{
    id: "js-loading",
    class: "loading",
    src: "img/loading-circle.gif",
    alt: "ローディング"
  }
});

const testMsg = createAttributedElements({
    tag:"p",
    attrObj:{
      class:"test-msg",
    },
    text:"テストメッセージ"
});
```

Reply:
I feel that it would be nice if you could give the function a name that shows what you created based on the function when you call it.

Can you try to apply this function to Mai-san's existing situation?

Also, it just occurred to me.
I thought it might be possible to use str since it would definitely contain a string. But this is IMO.


```js
//text→str
const createAttributedElements = ({tag,attrObj,str}) =>{
    const element = document.createElement(tag);
    Object.keys(attrObj).forEach((attribute) => {
        element.setAttribute(attribute, attrObj[attribute]);
    });
    if(str !== undefined) element.textContent = str;
    return element;
}
```

Reply:
Thank you for your opinion.

Certainly, if the function name is associated with the meaning of the argument content, 
it doesn't matter if there is no property.
I was convinced.

I will try to apply this function.

Reply:
I renamed the function and changed it to how to create an object with Object.keys.
https://github.com/nakagawamai/js-lesson/pull/19/commits/ebd59580c237e80d3a328175b055bf895c909c1f